source/clear-linux/guides/tooling/autospec.rst

```diff
@@ -322,6 +322,117 @@ and dependencies in order for autospec to complete a successful build.
+Machine)` allows testing is completely isolated environment. 
+
+To test an autospec created package inside a VM:
+
+#. Download the |CL| KVM image into the :file:`~/clearlinux` directory as
+   :file:`clear.img`. The location and name :file:`clear.img.xz` is important
+   for the tooling to work:
+
+   .. code-block:: bash
+
+      cd ~/clearlinux
+      curl -o clear.img.xz https://download.clearlinux.org/image/$(curl https://download.clearlinux.org/image/latest-images | grep '[0-9]'-kvm)
+
+#. Extract the downloaded |CL| KVM image:
+
+   .. code-block:: bash
+
+      unxz -v clear.img.xz
+
+#. Copy the QEMU start script and virtual firmware needed for KVM into the
+   :file:`~/clearlinux` directory:
+
+   .. code-block:: bash
+
+      cp ~/clearlinux/projects/common/start_qemu.sh .
+      cp /usr/share/qemu/OVMF.fd .
+
+#. Run :command:`make install` from the package's autospec directory. The
+   :command:`make install` command mounts the downloaded |CL| KVM image and
+   installs the autospec created RPM into it:
+
+   .. code-block:: bash
+
+      cd ~/clearlinux/packages/<package-name>
+      make install
+
+   The code that makes this possible can be viewed by searching for the
+   *install:*  target in the `Makefile.common file on GitHub`_.
+
+#. Return back to the :file:`~/clearlinux` directory and start the |CL| VM:
+
+   .. code-block:: bash
+
+      cd ~/clearlinux/
+      sudo ./start_qemu.sh clear.img
+
+#. A new |CL| VM will launch in the console. Log into the VM as *root* and set
+   a new pasword for the VM.
+
+#. Check that the software is installed in the |CL| VM as expected and perform
+   any relevant tests.
+
+#. After testing has been completed, the |CL| VM can be powered off and
+   deleted:
+
+   .. code-block:: bash
+      
+      poweroff 
+      rm clear.img
+
+
+Test directly on a development machine
+======================================
+
+The |CL| development tooling also includes a method to extract autospec
+created RPMs locally onto a |CL| development system for testing. Extracting a
+RPM directly onto a system  offers quicker testing, however conflicts may
+occur and responsibility to remove the software after testing is up to the
+developer.
+
+To test an autospec created package directly on the |CL| development system:
+
+#. Run :command:`make install-local` from the package's autospec directory.
+   The :command:`make install-local` command extracts the RPM directly onto
+   the filesystem of the running |CL| system:
+
+   .. code-block:: bash
+
+      cd ~/clearlinux/packages/<package-name>
+      make install-local
+
+   The code that makes this possible can be viewed by searching for the
+   *install-local:*  target in the `Makefile.common file on GitHub`_.
+
+#. Check that the software is installed as expected and perform any relevant
+   tests.
+
+#. After testing has been completed, the software and any related files must
+   be deleted manually.
```

Comment:
Manual deletion is okay, but additional files may have been added or modified as part of the `make install-local`. I would document that `swupd verify --fix --picky` is an easier option to perform all of this cleanup at once.

Reply:
@puneetse Any thoughts on the approach I describe here? Or do you want to keep this step written as-is right now?

Reply:
I like the `swupd verify --fix ....` command @phmccarty offers. I recommend adding a link to "Update integrity" header in this doc: https://clearlinux.org/documentation/clear-linux/concepts/swupd-about . Understanding how `swupd` works is critical to developer adoption. Can you try wrapping this up today? Would be very nice to have for next week.

Reply:
I was split on this and iterating on the wording. 

Basically, I want it to be abundantly clear that there is a responsibility involved to identify and remove any files, and ultimately that responsibility falls on anyone using individual RPMS on Clear. A callback to `swupd verify` would be nice but I don't want it may take away from the matter at hand.

Reply:
I've pushed a version I think works well. Let me know what you think.

Reply:
LGTM. Nice compromise. @phmccarty, are you OK with this?

Reply:
Yes, the new revision looks good to me. Thanks.